opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFromEntity.tsx

```diff
@@ -708,11 +708,14 @@ const StixCoreRelationshipCreationFromEntity: FunctionComponent<StixCoreRelation
   } = useEntityToggle(getLocalStorageKey(entityId));
 
   useEffect(() => {
-    const newTargetEntities: TargetEntity[] = Object.values(selectedElements).map((item) => ({
-      id: item.id,
-      entity_type: item.entity_type ?? '',
-      name: item.name ?? item.observable_value ?? item.representative?.main.toString() ?? '',
-    }));
+    const newTargetEntities: TargetEntity[] = Object.values(selectedElements).map((item) => {
+      const name = item.name ?? item.observable_value ?? item.attribute_abstract ?? item.explanation ?? item.representative?.main ?? '';
```

Comment:
Shouldn't it be the getMainRepresentative method used here ?

Reply:
good idea